### ISSUE TITLE
Type aliases don't work for constructing Tuple or Unit-like structs

### ISSUE THREAD

**Issue by Popog** [rust-lang/rust#37248]
Starting with what works:
```rust
struct Struct{a: i32, b: i32}
type StructAlias = Struct;

let _a1: Struct = Struct{a: 1, b: 2};
let _a2: StructAlias = StructAlias{a: 1, b: 2};
```

And comparing it to what doesn't:
```rust
struct Unit;
type UnitAlias = Unit;

let _b1: Unit = Unit;
let _b2: Unit = Unit{};
let _b3: UnitAlias = UnitAlias;  // unresolved name `UnitAlias`
let _b4: UnitAlias = UnitAlias{};

struct Tuple(i32);
type TupleAlias = Tuple;

let _c1: Tuple = Tuple(1);
let _c2: TupleAlias = TupleAlias(1); // unresolved name `TupleAlias`
```

Given that `a2` and `b4` compile, `b3` and `c2` should also compile (or at least give better error messages).

Also `d3` should give the same error as `d1`:
```rust
struct UnitBraces{};
type UnitBracesAlias = UnitBraces;

let _d1: UnitBraces = UnitBraces; // struct called like a function
let _d2: UnitBraces = UnitBraces{};
let _d3: UnitBracesAlias = UnitBracesAlias; // unresolved name `UnitBracesAlias`
let _d4: UnitBracesAlias = UnitBracesAlias{};
```

**Comment by petrochenkov** [comment]
This is a duplicate of few other issues (can't find them right now).
Here's an explanation of why type aliases don't work in expression/pattern positions: https://github.com/rust-lang/rfcs/blob/master/text/1506-adt-kinds.md
Better error messages are in my work queue (ETA: 1-2 months, maybe sooner).

**Comment by TimNN** [comment]
The best duplicate is #17422, I think.

**Comment by arielb1** [comment]
also similar to #26264, but less likely to be fixed.

**Comment by Popog** [comment]
So long as better error messages are on the way, that's good enough for me. I just ran into this and had no idea what the issue was and decided to investigate.

(P.S. I switched to `use X as Y;` which is what I should have been doing in the first place)
